presto-orc/src/main/java/com/facebook/presto/orc/reader/ListSelectiveStreamReader.java

```diff
@@ -0,0 +1,731 @@
+                    int positionsToSkip = 0;
+                    for (int j = 0; j < succeedingPositionsToFail; j++) {
+                        i++;
+                        int nextPosition = positions[i];
+                        positionsToSkip += 1 + nextPosition - streamPosition;
+                        streamPosition = nextPosition + 1;
+                    }
+                    skippedElements += skip(positionsToSkip);
+                }
+            }
+        }
+        elementOffsets[outputPositionCount] = elementPositionCount + skippedElements;
+
+        populateElementPositions(elementPositionCount);
+
+        if (listFilter != null) {
+            listFilter.populateElementFilters(outputPositionCount, nulls, elementLengths, elementPositionCount);
+        }
+
+        if (elementStreamReader != null) {
+            elementStreamReader.read(elementReadOffset, elementPositions, elementPositionCount);
+        }
+        elementReadOffset += elementPositionCount + skippedElements;
+
+        if (listFilter == null || level > 0) {
+            populateOutputPositionsNoFilter(elementPositionCount);
+        }
+        else {
+            populateOutputPositionsWithFilter(elementPositionCount);
+        }
+
+        return streamPosition;
+    }
+
+    private void populateElementPositions(int elementPositionCount)
+    {
+        elementPositions = ensureCapacity(elementPositions, elementPositionCount);
+
+        int index = 0;
+        for (int i = 0; i < outputPositionCount; i++) {
+            for (int j = 0; j < elementLengths[i]; j++) {
+                elementPositions[index] = elementOffsets[i] + j;
+                index++;
+            }
+        }
+    }
+
+    private void populateOutputPositionsNoFilter(int elementPositionCount)
+    {
+        if (outputRequired) {
+            elementOutputPositionCount = elementPositionCount;
+            elementOutputPositions = ensureCapacity(elementOutputPositions, elementPositionCount);
+            System.arraycopy(elementPositions, 0, elementOutputPositions, 0, elementPositionCount);
+
+            int offset = 0;
+            for (int i = 0; i < outputPositionCount; i++) {
+                offsets[i] = offset;
+                offset += elementLengths[i];
+            }
+            offsets[outputPositionCount] = offset;
+        }
+    }
+
+    private void populateOutputPositionsWithFilter(int elementPositionCount)
+    {
+        elementOutputPositionCount = 0;
+        elementOutputPositions = ensureCapacity(elementOutputPositions, elementPositionCount);
+
+        indexOutOfBounds = listFilter.getTopLevelIndexOutOfBounds();
+
+        int outputPosition = 0;
+        int elementOffset = 0;
+        boolean[] positionsFailed = listFilter.getTopLevelFailed();
+        for (int i = 0; i < outputPositionCount; i++) {
+            if (!positionsFailed[i]) {
+                // TODO How to raise an error if outputRequired == false
```

Comment:
Is this intentionally left here?

Reply:
@tdcmeehan Good catch. I addressed this TODO, but forgot to remove it. I'm removing it now and adding test cases.